crates/wit-component/src/validation.rs

```diff
@@ -172,29 +195,107 @@ pub enum Import {
     MainModuleMemory,
 
     /// An adapter is importing an arbitrary item from the main module.
-    MainModuleExport { name: String, kind: ExportKind },
+    MainModuleExport {
+        name: String,
+        kind: ExportKind,
+    },
 
     /// An arbitrary item from either the main module or an adapter is being
     /// imported.
     ///
     /// (should probably subsume `MainModule*` and maybe `AdapterExport` above
     /// one day.
     Item(Item),
+
+    ExportedTaskReturn(Function),
```

Comment:
Mind adding some comments to these variants as well? (nothing huge, just something indicating what it is like the above variants)